Switch from useIsSimpleSeller hook to getSiteOption

markbiek opened the pull request:
#### Changes proposed in this Pull Request

The hook isn't necessary. All we need to do is use `getSiteOption` to check if the site intent is `sell` on the WooCommerce landing page (the only place the hook was being used).

#### Testing instructions

Visit a non-seller website admin and go to `/woocommerce-installation`. You should see the following content:

![image](https://user-images.githubusercontent.com/917632/157089939-dd1d7120-6236-42b7-abd9-dddd101fe5f6.png)

Then visit the `/woocommerce-installation` page for a website that used the "Start simple" seller signup flow. You should see the following content instead:

![image](https://user-images.githubusercontent.com/917632/157090028-846fadd7-5e1e-4a10-8394-68e1d2875eee.png)

Related to #62031

markbiek committed:
Switch from useIsSimpleSeller hook to getSiteOption

The hook isn't necessary. All we need to do is use `getSiteOption` to check if the site intent is `sell` on the WooCommerce landing page (the only place the hook was being used).

nelsonec87 reviewed (approved):
Worked as described!

markbiek committed:
Remove useIsSimpleSeller hook (#62110)

Originally, we were using the hook to check both the site intent _and_ that the site plan _wasn't_ "ecommerce-bundle".

Checking the site plan isn't actually necessary since people with the full ecommerce plan will see the WooCommerce admin and not the landing page.

markbiek commented:
Thanks @nelsonec87 !

@andres-blanco Would you mind giving this a quick 👀 when you have a second? Just since you are familiar with the other, related PR?

andres-blanco commented on client/my-sites/woocommerce/landing-page.tsx:
```diff
@@ -38,7 +39,7 @@ const LandingPage: React.FunctionComponent< Props > = ( { siteId } ) => {
 	const { __ } = useI18n();
 	const navigationItems = [ { label: 'WooCommerce' } ];
 	const ctaRef = useRef( null );
-	const simpleSeller = useIsSimpleSeller();
```
I've seen this used differently.
https://github.com/Automattic/wp-calypso/blob/f2e6f88ae65c8571bbac21b284c29b48ec987598/client/my-sites/checkout/composite-checkout/components/prepurchase-notices/jetpack-plugin-required-version-notice.jsx#L47-L49

I think the current usage is fine but I find the one I linked a little more concise. Anyway I'll leave the choice up to you 😄 .

markbiek replied:
Oh, I like that. Much tidier! I'll get that updated shortly.

Thanks for the tip!

andres-blanco reviewed (approved):
This is working fine! I left a note but nothing blocking
